### ISSUE TITLE
docs: explicitly cover which APIs can be used in `--insecure` mode

### ISSUE THREAD

**Issue by TimJones** [siderolabs/talos#5311]
It is unclear currently which of the APIs/`talosctl` commands are available when a node is not configured, using `--insecure` flag.

**Comment by sanmai-NL** [comment]
@TimJones Is there any documentation on the necessity of ‘insecure’ (unauthenticated, it deserves a rename) bootstrapping the config (in maintainance mode)? Can it be avoided? What impact does it have when network-level security can't be guaranteed?

**Comment by steverfrancis** [comment]
It can be avoided with Omni (as SideroLink has a secure tunnel to Omni), but not with plain Talos - have to get the config onto the system somehow...

**Comment by sanmai-NL** [comment]
In the meantime I found out that some installation procedures do allow providing the machine config out-of-band/without networking, e.g. NoCloud with cloud-init.

**Comment by TimJones** [comment]
@sanmai-NL I don't think we have explicit docs covering it, but it's basically for when Talos is first booted and in maintenance mode, there is no PKI. Since there is no PKI we can't use the usual mTLS auth, and hence the `--insecure` flag for the few APIs that are required for Talos to transition *out* of maintenance mode. It is both unauthenticated *and* insecure (as there is no TLS layer).

**Comment by sanmai-NL** [comment]
In our case, the transport is system local on a single user system, hence my preference for unauthenticated over insecure in the general. Insecure is indeed a broad term that encompasses unauthenticated but possibly many other properties too, which gave me pause.

**Comment by rothgar** [comment]
This was fixed in #11430 

Available here https://www.talos.dev/latest/talos-guides/configuration/insecure/
